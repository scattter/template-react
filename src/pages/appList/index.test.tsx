import { fireEvent, render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import AppList from './index';

describe('test app list page', () => {
  const wrapper = shallow(<AppList />);

  it('should success render', function () {
    expect(wrapper).toMatchSnapshot();
  });

  it('should success go to app detail page by click li', function () {
    const history = createMemoryHistory({ initialEntries: ['/app'] });
    const { getByText } = render(
      <Router location={history.location} navigator={history}>
        <AppList />
      </Router>,
    );
    expect(history.location.pathname).toBe('/app');
    fireEvent.click(getByText('this is detail NO.1'));
    expect(history.location.pathname).toBe('/app/1');
  });
});
