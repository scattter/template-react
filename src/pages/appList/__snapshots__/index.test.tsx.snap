// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app list page should success render 1`] = `
<div>
  <span>
    this is app detail
  </span>
  <li>
    <Link
      to="/app/1"
    >
      this is detail NO.1
    </Link>
  </li>
  <li>
    <Link
      to="/app/2"
    >
      this is detail NO.2
    </Link>
  </li>
</div>
`;
